### PR TITLE
John 10 26

### PR DESCRIPTION
**Pull request by johnvaltix**
Jiras covered in this PR:
VAL-6569 - Custom IDS/IPS
VAL-6578 - Custom WAF
VAL-6575 (VAL-6129) - Log Forwarding CSP Account association
VAL-6598 - Service Principals and GLWB Endpoint Creation Gateway Settings

### CODE REVIEW

File: docs/resources/gateway.md [hunk 1]
```diff
@@ -306,7 +306,29 @@ settings {
...
+
+~> **Note on GWLB Endpoint (GWLBe) Creation**
+The GWLB Endpoint (GWLBe) creation default is set to `true` when Valtix orchestrates the GWLB when deploying an Egress Gateway. When not specified or explicitly set to `true`, Valtix orchestrates a GWLBe and connects it to the GWLB.  When set to `false`, Valtix will not orchestrate a GWLBe and will rely on the user to create any GWLBes using the AWS Terraform Provider or AWS Console and connect them to the GWLB.
+
+### To specify GWLB Service Principals
+```hcl
+settings {
+  name  = "controller.gateway.aws.gwlb.service_principals"
+  value = ["*"]
+}
```
Comment (hardikvaltix) [inline docs/resources/gateway.md:327]: @johnvaltix , can you update the example , syntax is not correct , here is the correct example : 
```
  settings {
  name  = "controller.gateway.aws.gwlb.service_principals"
  value = <<EOT
           *
           arn:aws:iam::902505820618:root
           arn:aws:iam::059126560514:root
           EOT
}
```

**Review by hardikvaltix — CHANGES_REQUESTED** [review]
please update  `controller.gateway.aws.gwlb.service_principals` example

**Review by hardikvaltix — COMMENTED** [review]
gateway settings documents looks good , @sunilvaltix can you please help to review custom rules document

File: docs/resources/profile_application_threat.md [hunk 2]
```diff
@@ -72,6 +92,9 @@ resource "valtix_profile_application_threat" "waf_manual" {
...
 * `delay_by_days_trustwave` - (Optional) Number of days to delay updating the Trustwave Ruleset version after it has been published by Valtix. Applicable values are integers from `0` to `30`.  A value of `0` means immediate update (0 days). The default value is `0` (immediately). New Rulesets as published as soon as updates are available from the Vendor and validation testing is completed by Valtix.
 * `trustwave_ruleset_version` - (Optional) Trustwave Ruleset version. Applicable values can be found from within the UI. The Rulesets are published frequently. Unless a specific version is desired, Valtix recommends using Auto Update as described above. If this argument is specified, Auto Update of Trustwave Ruleset is disabled and the Profile will use the specified Trustwave Ruleset version.
+* `auto_update_custom` - (Optional) Auto Update of the Custom Ruleset version with a delay specified by `delay_by_days_crs` argument. Applicable values are `true` or `false`.  If not specified, the default value is `true`.
```
Comment (sunilvaltix) [inline docs/resources/profile_application_threat.md:95]: The default for auto_update_custom is false.

File: docs/resources/profile_application_threat.md [hunk 3]
```diff
@@ -72,6 +92,9 @@ resource "valtix_profile_application_threat" "waf_manual" {
...
+* `delay_by_days_custom` - (Optional) Number of days to delay updating the Custom Ruleset version after it has been imported by the user. Applicable values are integers from `0` to `30`.  A value of `0` means immediate update (0 days). The default value is `0` (immediately).
+* `custom_ruleset_version` - (Optional) Custom Ruleset version. Applicable values can be found from within the UI as specified by the user when defining and importing the Custom Rulesets. Unless a specific version is desired, Valtix recommends using Auto Update. If this argument is specified, Auto Update of Custom Ruleset is disabled and the Profile will use the specified CRS Ruleset version.
```
Comment (sunilvaltix) [inline docs/resources/profile_application_threat.md:97]: > use the specified CRS Ruleset version.

"use the specified Custom Ruleset version"

File: docs/resources/profile_network_intrusion.md [hunk 4]
```diff
@@ -78,8 +81,11 @@ resource "valtix_profile_network_intrusion" "ips_manual" {
...
+* `delay_by_days` - (Optional) Number of days to delay applying the Talos Network Intrusion Ruleset version after it has been published by Valtix. Applicable values are integers from `0` to `30`.  A value of `0` means immediate update (0 days). The default value is `0` (immediately). New Rulesets as published as soon as updates are available from the Vendor and validation testing is completed by Valtix.
+* `talos_ruleset_version` - (Optional) Talos Network Intrusion Ruleset version. Applicable values can be found from within the UI. The Rulesets are published frequently. Unless a specific version is desired, Valtix recommends using Auto Update. If this argument is specified, Auto Update of Talos Network Intrusion Ruleset is disabled and the Profile will use the specified Talos Network Intrusion Ruleset version.
+* `auto_update_custom` - (Optional) Auto Update the Custom Network Intrusion Ruleset version. Valid values are `true` or `false`.  Default (if unspecified) is `true`.
```
Comment (sunilvaltix) [inline docs/resources/profile_network_intrusion.md:86]: The default is false.

File: docs/resources/gateway.md (hunk 1 above)
Comment (kiranvaltix) [reply docs/resources/gateway.md:327]: @hardikvaltix @johnvaltix can you please confirm this works. Ideally the EOT on the last line must not have any spaces (no indentation). But please check it works

File: docs/resources/csp_log_association.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# valtix_csp_log_association
```
Comment (kiranvaltix) [inline docs/resources/csp_log_association.md:1]: The resource file name in the documentation says csp_log_association, but the resource name says valtix_cloud_account_log_profile_assocation. These 2 are not matching. Can you please look into this

If the resource is valtix_cloud_account_log_profile_assocation, then shouldnt the file name be cloud_account_log_profile_assocation?

File: docs/resources/gateway.md (hunk 1 above)
Comment (hardikvaltix) [reply docs/resources/gateway.md:327]: @kiranvaltix  yes its works fine , there is no space

**Pull request merged**